src/viadot/sources/eurostat.py

```diff
@@ -0,0 +1,290 @@
+        """Function for creating DataFrame from JSON pulled from Eurostat.
+        Returns:
+            pd.DataFrame: With 4 columns: index, geo, time, indicator.
+        """
+
+        class T_SIGNAL:
+            signal_keys_list: list
+            signal_index_list: list
+            signal_label_list: list
+            signal_name: str
+
+        # Dataframe creation
+        columns0 = signals[0].copy()
+        columns0.append("indicator")
+        df = pd.DataFrame(columns=columns0)
+        indicator_list = []
+        index_list = []
+        signal_lists = []
+
+        # get the dictionary from the inputs
+        eurostat_dictionary = signals[-1]
+
+        for signal in signals[0]:
+            signal_struct = T_SIGNAL()
+            signal_struct.signal_name = signal
+            signal_struct.signal_keys_list = list(
+                eurostat_dictionary["dimension"][signal]["category"]["index"].keys()
+            )
+            signal_struct.signal_index_list = list(
+                eurostat_dictionary["dimension"][signal]["category"]["index"].values()
+            )
+            signal_label_dict = eurostat_dictionary["dimension"][signal]["category"][
+                "label"
+            ]
+            signal_struct.signal_label_list = [
+                signal_label_dict[i] for i in signal_struct.signal_keys_list
+            ]
+            signal_lists.append(signal_struct)
+
+        col_signal_temp = []
+        row_signal_temp = []
+        for row_index, row_label in zip(
+            signal_lists[0].signal_index_list, signal_lists[0].signal_label_list
+        ):  # rows
+            for col_index, col_label in zip(
+                signal_lists[1].signal_index_list, signal_lists[1].signal_label_list
+            ):  # cols
+                index = str(
+                    col_index + row_index * len(signal_lists[1].signal_label_list)
+                )
+                if index in eurostat_dictionary["value"].keys():
+                    index_list.append(index)
+                    col_signal_temp.append(col_label)
+                    row_signal_temp.append(row_label)
+
+        indicator_list = [eurostat_dictionary["value"][i] for i in index_list]
+        df.indicator = indicator_list
+        df[signal_lists[1].signal_name] = col_signal_temp
+        df[signal_lists[0].signal_name] = row_signal_temp
+
+        return df
+
+    @add_viadot_metadata_columns
+    def to_df(
+        self,
+        dataset_code: str,
+        params: dict = None,
+        base_url: str = "https://ec.europa.eu/eurostat/api/dissemination/statistics/1.0/data/",
+        requested_columns: list = None,
+        if_empty: str = "fail",
+        tests: dict = None,
+    ) -> pd.DataFrame:
+        """Function responsible for getting response, creating DataFrame using method 'eurostat_dictionary_to_df'
+           with validation of provided parameters and their codes if needed.
```

Comment:
Sry just noticed it here but -- by "codes", you mean "currency codes", right? If so, can you use that full form in the documentation (and consider also using it in variable names)? It would make it easier to understand.

Reply:
No, codes reffer to `codes` of specific parmeter e.g (for dataset "prc_hicp_midx" we could filetr dataset by specific item. "Food and non-alcoholic beverages" has code "CP01" - and this code is needed for filtering the data) in dataset that could be used to retriev specific values from dataset. It can be currency code as well, but not only.

Reply:
Ah, okay. Is there a list of these codes you could link to in the docstring, or is this info only in your head? Or at least sth explaining what the codes are -- just like you explained to me above? Since this info is not possible to understand from the code itself.